src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,139 @@
+            INSTANCE = new Shooter();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * State space configuration function. Note that there are 2 different configurations.
+     *
+     * @param isInertiaBased is the configuration for the state space.
+     *                       If the value is true, the state space is based off of an inertia model.
+     *                       Otherwise, the state space is based off of a voltage equation model.
+     * @return the linear system loop (based on the type).
+     */
+    private LinearSystemLoop<N1, N1, N1> configStateSpace(boolean isInertiaBased) {
+        final DCMotor motor = DCMotor.getFalcon500(1);
+
+        LinearSystem<N1, N1, N1> flywheel_plant;
+        if (isInertiaBased) {
+            flywheel_plant = LinearSystemId.createFlywheelSystem(motor, J, GEAR_RATIO);
+        } else {
+            flywheel_plant = LinearSystemId.identifyVelocitySystem(Kv, Ka);
+        }
+
+        LinearQuadraticRegulator<N1, N1, N1> quadraticRegulator = new LinearQuadraticRegulator<>(
+                flywheel_plant,
+                VecBuilder.fill(MODEL_TOLERANCE),
+                VecBuilder.fill(SENSOR_TOLERANCE),
+                LOOP_PERIOD);
+        quadraticRegulator.latencyCompensate(flywheel_plant, LOOP_PERIOD, TALON_TIMEOUT / 1000.0);
+
+        KalmanFilter<N1, N1, N1> kalmanFilter = new KalmanFilter<>(
+                Nat.N1(), Nat.N1(),
+                flywheel_plant,
+                VecBuilder.fill(MODEL_TOLERANCE),
+                VecBuilder.fill(SENSOR_TOLERANCE),
+                LOOP_PERIOD);
+
+        return new LinearSystemLoop<>(
+                flywheel_plant,
+                quadraticRegulator,
+                kalmanFilter,
+                NOMINAL_VOLTAGE, LOOP_PERIOD);
+    }
+
+    /**
+     * Gets the velocity of the motor.
+     *
+     * @return the velocity of the motor. [rps]
+     */
+    public double getVelocity() {
+        return unitModel.toVelocity(motor.getSelectedSensorVelocity());
+    }
+
+    /**
+     * Sets the velocity of the motor.
+     *
+     * @param velocity is the velocity setpoint. [rps]
+     */
+    public void setVelocity(double velocity) {
+        linearSystemLoop.setNextR(VecBuilder.fill(velocity));
+        linearSystemLoop.correct(VecBuilder.fill(getVelocity()));
+        linearSystemLoop.predict(currentTime - lastTime);
+
+        motor.setVoltage(MathUtil.clamp(linearSystemLoop.getU(0), -NOMINAL_VOLTAGE, NOMINAL_VOLTAGE));
+    }
+
+    public void setPower(double output) {
+        motor.set(ControlMode.PercentOutput, output);
+    }
+
+    /**
+     * Calculates the velocity setpoint according to the distance from the target.
+     * Once the data from the shooter is acquired this function will be changed.
+     *
+     * @param distance is the distance from the target. [m]
+     * @return 15. [rps]
+     */
+    public double getSetpointVelocity(double distance) {
+        return 15 * distance;
```

Comment:
Why 15?
I guess it's related to some constant but which one.

Reply:
It has no meaning. Just a random number that I came up with.